File: ControleGastos/ControleGastos.Core/Models/Transacao.cs
using ControleGastos.Core.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ControleGastos.Core.Models
{
    [Table("Transacoes")]
    public class Transacao
    {
        [Key]
        public int Id { get; set; }

        [MaxLength(400)]
        public String Descricao { get; set; }

        public decimal Valor { get; set; }
        
        public TipoTransacao Tipo { get; set; }

        public int CategoriaId { get; set; }

        public int PessoaId { get; set; }
    }
}

File: ControleGastosFront/src/Models/Transacao.ts
type TransacaoJson = {
  id?: number
  descricao: string
  valor: number
  tipo: number
  categoriaId: number
  pessoaId: number
}

export default class Transacao {
  id?: number
  descricao: string
  valor: number
  tipo: number
  categoriaId: number
  pessoaId: number

  constructor(
    descricao: string,
    valor: number,
    tipo: number,
    categoriaId: number,
    pessoaId: number,
    id?: number
  ) {
    this.id = id
    this.descricao = descricao
    this.valor = valor
    this.tipo = tipo
    this.categoriaId = categoriaId
    this.pessoaId = pessoaId
  }

  static fromJson(json: TransacaoJson): Transacao {
    return new Transacao(
      json.descricao,
      json.valor,
      json.tipo,
      json.categoriaId,
      json.pessoaId,
      json.id
    )
  }

  toJson(): TransacaoJson {
    return {
      id: this.id,
      descricao: this.descricao,
      valor: this.valor,
      tipo: this.tipo,
      categoriaId: this.categoriaId,
      pessoaId: this.pessoaId
    }
  }
}